### PR TITLE
isw: init at 0.3.0

### PR DESCRIPTION
**Pull request by thom-cameron**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by thom-cameron** [comment]
ofborg-eval doesn't seem to be doing anything. Trying to get it working with:

@grahamcofborg eval

File: pkgs/tools/misc/isw/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  fetchFromGitLab,
+  rustPlatform,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "isw";
+  version = "0.3.0";
+
+  src = fetchFromGitLab {
+    owner = "thom-cameron";
+    repo = pname;
```
Comment (Aleksanaa) [inline pkgs/tools/misc/isw/default.nix:12]:
```suggestion
    repo = "isw";
```

File: pkgs/tools/misc/isw/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  };
+
+  cargoHash = "sha256-LpKiP7bwiNEUXYvQsUG//wqOwGmg2UD3DA0Gb2h7kMw=";
+
+  meta = with lib; {
+    description = "a simple terminal stopwatch application";
+    homepage = "https://gitlab.com/thom-cameron/isw";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ thom-cameron ];
+  };
```
Comment (Aleksanaa) [inline pkgs/tools/misc/isw/default.nix:24]:
```suggestion
  meta = {
    description = "Simple terminal stopwatch application";
    homepage = "https://gitlab.com/thom-cameron/isw";
    license = lib.licenses.gpl3Only;
    maintainers = with lib.maintainers; [ thom-cameron ];
  };
```

Comment (Aleksanaa) [reply pkgs/tools/misc/isw/default.nix:24]: And missing `mainProgram`

**Review by gepbird — CHANGES_REQUESTED** [review]
This package is not called anywhere, so for example `nix-build -A isw` doesn't work. To fix this, you could put  `isw = callPackage ../tools/misc/isw { };` into `pkgs/top-level/all-packages.nix`.
However for new packages `by-name` is preferred, where you don't need to `callPackage`, please move the nix file into `pkgs/by-name/is/isw/package.nix.

Personally I like this package, I've been looking for a simple and clean TUI stopwatch. For now I've used termdown but I didn't really like it as it didn't show sub-second time.
However as per https://github.com/NixOS/nixpkgs/tree/master/pkgs#quick-start-to-adding-a-package I'm afraid this project is not mature enough for nixpkgs, it was created 1 week ago, and has 1 stars.

**Comment by thom-cameron** [comment]
That makes sense and has helped me understand how nixpkgs works better, thank you. 

Glad you like the look of the app and no worries about it not being mature enough. I'll work on it or a while longer, hopefully some other people will start finding it useful, then I'll open a PR that works properly.

Cheers

**Pull request closed without merging**